File: backend/src/main/java/todoktodok/backend/member/application/dto/response/MemberResponse.java
package todoktodok.backend.member.application.dto.response;

import todoktodok.backend.member.domain.Member;

public record MemberResponse(
        Long memberId,
        String nickname,
        String profileImage
) {

    private static final String DELETED_MEMBER_PROFILE_IMAGE = "https://todoktodok.s3.ap-northeast-2.amazonaws.com/todoktodok-images/profile/todoki.png";
    private static final String DELETED_MEMBER_NICKNAME = "(알수없음)";

    public MemberResponse(final Member member) {
        this(
                member.getId(),
                getNickname(member),
                getProfileImage(member)
        );
    }

    public static String getNickname(final Member member) {
        return member.isDeleted() ? DELETED_MEMBER_NICKNAME : member.getNickname();
    }

    public static String getProfileImage(final Member member) {
        return member.isDeleted() ? DELETED_MEMBER_PROFILE_IMAGE : member.getProfileImage();
    }
}

File: android/TodokTodok/app/src/main/java/com/team/todoktodok/data/network/response/discussion/MemberResponse.kt
package com.team.todoktodok.data.network.response.discussion

import com.team.domain.model.member.User
import kotlinx.serialization.Serializable

@Serializable
data class MemberResponse(
    val memberId: Long,
    val nickname: String,
    val profileImage: String,
)

fun MemberResponse.toDomain() = User(id = memberId, nickname = nickname, profileImage = profileImage)
